core/types/calls.cc

```diff
@@ -2175,6 +2175,47 @@ class Magic_selfNew : public IntrinsicMethod {
     }
 } Magic_selfNew;
 
+class Magic_splat : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
+        if (args.args.size() != 1) {
+            return;
+        }
+
+        auto &arg = args.args[0];
+        if (auto *tuple = cast_type<TupleType>(arg->type)) {
+            res.returnType = arg->type;
+            return;
+        }
+
+        if (auto *ap = cast_type<AppliedType>(arg->type)) {
+            if (ap->klass == Symbols::Array() || ap->klass.data(gs)->derivesFrom(gs, Symbols::Array())) {
+                res.returnType = arg->type;
+                return;
+            }
+        }
```

Comment:
Since both `TupleType` and `Array` have `to_a` sigs, I think neither of these cases are necessary.